File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int solution(string t, string p) {
    int answer = 0;
    
        for(int i=0; i<t.size()- p.size()+1; i++){
        string str = t.substr( i ,p.size());
        if (str<= p){
            cout << str << endl;
            answer++;
        }
    }
    
    return answer;
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int answer = 0;
        for(int i=0; i<t.length()-p.length()+1; i++ ){
            
            long num = Long.parseLong(t.substring(i, i+p.length()));
            System.out.println(num);
            
            if( num <= Long.parseLong(p) ){
                answer++;
            }
            
        }
        return answer;        
        
        
        
        
        
        
        
        
//         int answer = 0;
        
//         System.out.println(t.length());
//         System.out.println(p.length());
        
        
//         for(int i=0; i<t.length()-p.length()+1; i++){
            
//             StringBuilder sb = new StringBuilder();
            
//             for(int j=0; j< p.length(); j++){
//                 sb.append(t.charAt(j+i));
//             }
            
//             System.out.println("sb: " + sb.toString());

            
            
//             if( Long.parseLong(sb.toString()) <= Long.parseLong(p)){
//                 answer++;
//             }
            
            
            
            
            
//         }
        
        
        
//         return answer;
    }
}



